src/Archiver/CsvArchiver.php

```diff
@@ -131,16 +131,16 @@ public function successful(): void
     private function ensureDirectoryExists(string $directory)
     {
         if (!$this->filesystem->isExists($directory)) {
-            $this->filesystem->createDirectory($directory, 0777);
+            $this->filesystem->createDirectory($directory, $this->config->get('directory_permissions'));
         }
     }
 
-    private function newName(\SplFileObject $file): string
+    protected function newName(\SplFileObject $file): string
```

Comment:
why this is made protected?

Reply:
sorry this was meant to be in another PR.
this change is required to override new filename format of archived files through Preference. I didn't wanted to change it to public to allow change of name through Plugin.
usecase of this is that we need to put files into daily directories under archive folder and it can be easily achieved by this small harmless change

Reply:
Ok protected for library is ok